### PR TITLE
Update github.com/cespare/xxhash/v2 to v2.1.2

### PR DESCRIPTION
**Pull request by fedepaol**
github.com/cespare/xxhash/v2 v2.1.1 is incompatible with Go 1.17 (see https://github.com/cespare/xxhash/issues/54)
Here we bump up the dependency to solve the issue.

### CODE REVIEW

**Comment by fedepaol** [comment]
cc @zshi-redhat @SchSeba

File: hack/tools.go [hunk 1]
```diff
@@ -1,3 +1,4 @@
+//go:build tools
```
Comment (zshi-redhat) [inline hack/tools.go:1]: Is this change still needed?

Comment (fedepaol) [reply hack/tools.go:1]: Yes, that's why I pushed 3 times :-)

CI was complaining about it

Comment (zshi-redhat) [reply hack/tools.go:1]: wondering if we can remove the following line`+build tools`.

Comment (fedepaol) [reply hack/tools.go:1]: I think we can't. CI compares the generated file with what's in this PR and complains if they are different.

Comment (fedepaol) [reply hack/tools.go:1]:
```
diff --git a/hack/tools.go b/hack/tools.go
index 7a21725c..d1427a67 100644
--- a/hack/tools.go
+++ b/hack/tools.go
@@ -1,4 +1,4 @@
-// +build tools
+//go:build tools
```

This is if I remove the other line

**Review by zshi-redhat — APPROVED** [review]
